libraries/bsg_manycore_cuda.cpp

```diff
@@ -538,8 +538,13 @@ int hb_mc_device_init (hb_mc_device_t *device,
         XMALLOC(device->mc);
         *(device->mc) = {0};
 
-        BSG_MANYCORE_CALL(device->mc, hb_mc_manycore_init(device->mc, name, id))
-
+        //BSG_MANYCORE_CALL(device->mc, hb_mc_manycore_init(device->mc, name, id))
```

Comment:
@mrutt92 Out of curiosity, why do we replace `BSG_MANYCORE_CALL` with explicit return value checking?

Reply:
BSG_MANYCORE_CALL relies on a string internal the manycore object being set. But if the init function failed, the string may not be set. Passing an unset string to printf and friends is unsafe.

Reply:
Thanks @mrutt92 !